cli/class-block-plugin-checker.php

```diff
@@ -0,0 +1,476 @@
+<?php
+namespace WordPressdotorg\Plugin_Directory\CLI;
+
+use WordPressdotorg\Plugin_Directory\Readme\Parser as Readme_Parser;
+use WordPressdotorg\Plugin_Directory\Tools\Filesystem;
+use WordPressdotorg\Plugin_Directory\Tools\SVN;
+use WordPressdotorg\Plugin_Directory\Block_JSON\Parser as Block_JSON_Parser;
+use WordPressdotorg\Plugin_Directory\Block_JSON\Validator as Block_JSON_Validator;
+
+/**
+ * A class that can examine a plugin, and evaluate and return status info that would be useful for a validator or similar tool.
+ *
+ * Note: I've written this as one class for convenience, but as it's evolved I think it would make sense to split it in two: one for collecting data
+ * (particularly the prepare*() and find*() functions), and a separate class for the checks.
+ *
+ * @package WordPressdotorg\Plugin_Directory\CLI
+ */
+class Block_Plugin_Checker {
+
+	protected $allowed_hosts = array(
+		'github.com',
+		'plugins.svn.wordpress.org',
+	);
+
+	protected $path_to_plugin = null;
+	protected $check_methods = array();
+	protected $results = array();
+
+	public $slug = null;
+	public $repo_url = null;
+	public $repo_revision = null;
+	protected $readme_path = null;
+	protected $readme = null;
+	protected $headers = null;
+	protected $blocks = null;
+	protected $block_json_files = null;
+	protected $block_json = array();
```

Comment:
```suggestion
	protected $block_json_validation = array();
```

I think you meant `$block_json_validation` here? There are three places in the code that use it, assuming it's an array, but no places in the code that use just `block_json`...

Reply:
Good catch, thanks!